src/sentry/rules/conditions/event_frequency.py

```diff
@@ -125,3 +127,76 @@ def query_hook(self, event, start, end, environment_id):
             environment_id=environment_id,
             use_cache=True,
         )[event.group_id]
+
+
+percent_intervals = {
+    "1m": ("one minute", timedelta(minutes=1)),
+    "5m": ("five minutes", timedelta(minutes=5)),
+    "10m": ("ten minutes", timedelta(minutes=10)),
+    "30m": ("30 minutes", timedelta(minutes=30)),
+    "1h": ("one hour", timedelta(minutes=60)),
+}
+
+
+class EventFrequencyPercentForm(EventFrequencyForm):
+    intervals = percent_intervals
+    interval = forms.ChoiceField(
+        choices=[
+            (key, label)
+            for key, (label, duration) in sorted(
+                percent_intervals.items(),
+                key=lambda key____label__duration: key____label__duration[1][1],
+            )
+        ]
+    )
+    value = forms.IntegerField(widget=forms.TextInput())
+
+
+class EventFrequencyPercentCondition(BaseEventFrequencyCondition):
+    label = "The issue has more errors than {value} percent of sessions in {interval}"
+
+    def __init__(self, *args, **kwargs):
+        self.intervals = percent_intervals
+        self.form_cls = EventFrequencyPercentForm
+        super().__init__(*args, **kwargs)
+
+    def query_hook(self, event, start, end, environment_id):
+        project_id = event.project_id
+        cache_key = f"r.c.spc:{project_id}-{environment_id}-{start.replace(minute=0, second=0, microsecond=0, tzinfo=None)}-{end.replace(minute=0, second=0, microsecond=0, tzinfo=None)}".replace(
```

Comment:
You didn't actually change this to use the interval. If you take Dan's idea of querying for the last hour of data and dividing, it might save snuba queries as well because there wouldn't be a separate cache for each time interval. Then in your percentage calculation below you would have to account for the fact that session_count is for a full hour.

Reply:
Just pushed up an update using this idea. I was thinking of storing `end` without minutes, but I think it's okay since this function is only ever called with end basically always being the current time.

So now I'm averaging out the last hour of data and saving the extra snuba queries.